### DISCUSSION TITLE
Add guidelines to initial research query

### DISCUSSION THREAD

**Discussion by leitdeux** [assafelovic/gpt-researcher#482] [Ideas]
In the multi-agent workflow,  it would be great to be able to define guidelines not only for the writing of the report, but also a set of guidelines for the research to be performed. For example, the query could be "Find the latest financial news on Fortune 500 companies", but the guidelines could add some constraints, such as, "Do not include Apple and Meta in your research findings."

An example schema for `task.json` might look like:
```json
{
  "query": {
    "text": "Find the latest financial news on Fortune 500 companies",
    "guidelines": [
     "Exclude Apple and Meta from your findings", 
     "Only consider news articles written by British sources."
    ]
  }
}
```

**Comment by assafelovic** [comment]
Hey @leitdeux thanks for the feedback, have you considered simply adding these guidelines to the query? This way the `guidelines` field will be specific for writing the report

**Reply by leitdeux** [reply]
Thank you for the suggestion, I hadn't caught that the agent was indeed marking those specific keywords to be ignored in the web search queries it performed in the example I gave. Great work!
